operator/internal/webhook/admission/pgs/validation/podgangset_test.go

```diff
@@ -23,488 +23,157 @@ import (
+// Temporary helper function for remaining tests - to be refactored
 func createDummyPodGangSet(name string) *grovecorev1alpha1.PodGangSet {
-	return &grovecorev1alpha1.PodGangSet{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: "default",
-		},
-		Spec: grovecorev1alpha1.PodGangSetSpec{
-			Replicas: 1,
-			Template: grovecorev1alpha1.PodGangSetTemplateSpec{
-				TerminationDelay: &metav1.Duration{Duration: 30 * time.Second},
-				StartupType:      ptr.To(grovecorev1alpha1.CliqueStartupTypeAnyOrder),
-				Cliques: []*grovecorev1alpha1.PodCliqueTemplateSpec{
-					{
-						Name:        "test",
-						Labels:      nil,
-						Annotations: nil,
-						Spec: grovecorev1alpha1.PodCliqueSpec{
-							Replicas:     1,
-							RoleName:     "dummy-role",
-							MinAvailable: ptr.To[int32](1),
-						},
-					},
-				},
-			},
-		},
-	}
+	return testutils.NewPodGangSetBuilder(name, "default").
+		WithReplicas(1).
+		WithTerminationDelay(30 * time.Second).
+		WithCliqueStartupType(ptr.To(grovecorev1alpha1.CliqueStartupTypeAnyOrder)).
+		WithPodCliqueTemplateSpec(
+			testutils.NewPodCliqueTemplateSpecBuilder("test").
+				WithReplicas(1).
+				WithRoleName("dummy-role").
+				WithMinAvailable(1).
+				Build()).
+		Build()
 }
 
-// createDummyPodCliqueTemplate Helper function to create a valid PodCliqueTemplateSpec with predefined values
-// This is used to test the validation logic without needing a full setup.
-// It creates a PodCliqueTemplateSpec with a single container and minimal configuration.
 func createDummyPodCliqueTemplate(name string) *grovecorev1alpha1.PodCliqueTemplateSpec {
-	return &grovecorev1alpha1.PodCliqueTemplateSpec{
-		Name: name,
-		Spec: grovecorev1alpha1.PodCliqueSpec{
-			Replicas:     1,
-			MinAvailable: ptr.To(int32(1)),
-			PodSpec: corev1.PodSpec{
-				Containers: []corev1.Container{
-					{
-						Name:  "test-container",
-						Image: "test:latest",
-					},
-				},
-			},
-			RoleName: fmt.Sprintf("dummy-%s-role", name),
-		},
-	}
+	return testutils.NewPodCliqueTemplateSpecBuilder(name).
+		WithReplicas(1).
+		WithRoleName(fmt.Sprintf("dummy-%s-role", name)).
+		WithMinAvailable(1).
+		Build()
 }
 
-// Helper function to create a PodCliqueScalingGroupConfig
 func createScalingGroupConfig(name string, cliqueNames []string) grovecorev1alpha1.PodCliqueScalingGroupConfig {
 	return grovecorev1alpha1.PodCliqueScalingGroupConfig{
 		Name:        name,
 		CliqueNames: cliqueNames,
 	}
 }
 
-func TestPodCliqueTemplateNameValidation(t *testing.T) {
+func TestResourceNamingValidation(t *testing.T) {
 	testCases := []struct {
-		description     string
-		pgsName         string
-		cliqueNames     []string
-		expectError     bool
-		expectedErrMsg  string
-		expectedErrPath string
+		description      string
+		pgsName          string
+		cliqueNames      []string
+		scalingGroups    []grovecorev1alpha1.PodCliqueScalingGroupConfig
+		expectError      bool
+		expectedErrMsg   string
```

Comment:
Outside the scope of this PR as it wasn't introduced in it, but I'd recommend replacing string-based error testing with error codes and constants to decouple test logic from error message formatting, ensuring tests remain valid when messages are updated while providing a stable contract for error identification.